main.go

```diff
@@ -349,20 +368,55 @@ func BackoffOnError(err error) bool {
 	return false
 }
 
+func (h *HauserService) Init() error {
+	// Initialize the warehouse's schema
+	if err := h.warehouse.EnsureCompatibleExportTable(); err != nil {
+		return err
+	}
+	// NB: We SHOULD fetch the table columns ONLY after the call to EnsureCompatibleExportTable.
+	// The EnsureCompatibleExportTable function potentially alters the schema of the export table in the client warehouse.
+	h.tableColumns = h.warehouse.GetExportTableColumns()
+	for i, col := range h.tableColumns {
+		h.tableColumns[i] = strings.ToLower(col)
+	}
+	return nil
+}
+
+func (h *HauserService) ProcessNext() (int, error) {
+	lastSyncedRecord, err := h.warehouse.LastSyncPoint()
+	if err != nil {
+		return 0, err
+	}
+	return h.ProcessExportsSince(lastSyncedRecord)
+}
+
+func (h *HauserService) Run() {
+	if err := h.Init(); err != nil {
+		log.Fatal(err)
+	}
+	for {
+		if numBundles, err := h.ProcessNext(); err != nil {
+			if h.BackoffOnError(err) {
```

Comment:
I think you want to unconditionally call `BackoffOnError` here so that it resets when there is no error.

Reply:
Ah, yes. Of course!